### PR TITLE
Add Spinner component

### PR DESCRIPTION
**Pull request by unindented**
A spinner is an outline of a circle which animates around itself indicating to the user that things are processing. It is shown when we're unsure how long a task will take.

#### Pull request checklist
- [x] Component readme is up-to-date
- [x] UX Designer has approved visual changes

### CODE REVIEW

File: src/util/enums/spinner.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/*! Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license. */
+import { Enum } from 'typescript-string-enums';
```
Comment (unindented) [inline src/util/enums/spinner.ts]: Now that TypeScript supports string enums, we can remove this in a separate PR.

File: src/stories/assets/stories.css [hunk 2]
```diff
@@ -0,0 +1,4 @@
+.animations-disabled * {
```
Comment (unindented) [inline src/stories/assets/stories.css]: Suggestions for a better class name?

File: src/components/Spinner/index.tsx [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+export interface SpinnerProps extends BaseComponentProps {
+  /**
+   * Label displayed below the spinner.
+   */
+  label?: string;
+  /**
+   * LARGE: 48px, MEDIUM: 40px, SMALL: 32px, XSMALL: 24px. Defaults to medium.
+   */
+  size?: SpinnerSize;
```
Comment (unindented) [inline src/components/Spinner/index.tsx]: Waiting for Design to come back with all scenarios we need to support.

After that, I'll consult with @BrendanMcK, see how we want to deal with labels and ARIA.

File: src/components/Spinner/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+```
+<Spinner size={ SpinnerSize.large } />
+```
+
+### Spinner with label
+
+```
+<Spinner label="Loading..." />
+```
```
Comment (unindented) [inline src/components/Spinner/README.md]: @swese44 are these examples too obvious? What do we usually have as examples?

Comment (swese44) [reply src/components/Spinner/README.md]: i think that's fine, there's not much logic going into a spinner

**Comment by unindented** [comment]
Visual diffs are randomly failing one or both of the spinner tests. Seems like animations are not being disabled?

File: src/components/Spinner/README.md [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+---
+
+### Properties
+
+| Prop | Type | Description |
+| ---- | ---- | ----------- |
+| label | string | Text that will be rendered below the spinner.
+| size | SpinnerSize | Size of spinner.
+
+* indicates property is required
```
Comment (annr) [inline src/components/Spinner/README.md]: If there are no required properties we can remove this note.

**Comment by unindented** [comment]
The increased thresholds for spinners seems to have solved the issue.

File: src/components/Spinner/README.md (hunk 5 above)
Comment (unindented) [reply src/components/Spinner/README.md]: Fixed.

**Comment by unindented** [comment]
I'll still wait for Design to let us know all the different scenarios, cause I'll probably need to update things.

**Comment by unindented** [comment]
@mamuso @annr I've updated according to spec. Please review.

**Comment by unindented** [comment]
@adrianmg in case @mamuso is not available...

**Pull request merged**